### ISSUE TITLE
Adding another taxonomy...

### ISSUE THREAD

**Issue by infracritical** [MISP/misp-taxonomies#13]
OK, so I know about RTFM, but have read, re-read, re-re-read, et. al -- the documentation to determine if I was loosing it, or not.

I added a new directory under the /var/www/MISP/app/files/taxonomies directory, and clicked on 'Update Taxonomies" -- nothing.  Nothing showed up.

Am I missing something here?  Made it the same permissions as everything else.  The taxonomy was duplicated using the TLP as its 'base' (so that should've worked).  So...the question is -- what did I do wrong?

**Comment by adulau** [comment]
It should work. Maybe you didn't change the name space in your JSON file? I hope this helps.

**Comment by iglocska** [comment]
Just to make sure, basically what is needed is the following:

/var/www/MISP/app/files/taxonomies/my_taxonomy directory
/var/www/MISP/app/files/taxonomies/my_taxonomy/machinetag.json file containing the taxonomy data

The new taxonomy vocabulary has to have its own unique namespace, it needs a version number and a description at the very least.

Make sure that all of it is readable by the apache user and it should work.

If you've tried all of these and are still stuck, let us know!

**Comment by infracritical** [comment]
OK...a look at the constructs of the file:

{
  "namespace": "cipsectors",
  "expanded": "Critical Infrastructure Protection (CIP) Sectors",
  "description": "The Critical Infrastructure Protection (CIP) Sectors represent critical infrastructures and/or key resources that are vital to a given country. This model is U.S.-specifically related, as outlined and identified by the United States Department of Homeland Security (DHS) per Homeland Security Presidential Directive No. 7 (HSPD-7). URL: https://www.dhs.gov/critical-infrastructure-sectors",
  "version": 1,
  "predicates": [
    {
      "value": "chemical",
      "expanded": "(CHEMICAL SECTOR) The U.S. Department of Homeland Security is designated as the Sector-Specific Agency (SSA) for the Chemical Sector."
    },
    {
      "value": "commercial",
      "expanded": "(COMMERCIAL FACILITIES SECTOR) The U.S. Department of Homeland Security is designated as the Sector-Specific Agency (SSA) for the Commercial Facilities Sector."
    },
    {
      "value": "communications",
      "expanded": "(COMMUNICATIONS SECTOR) The Communications Sector is an integral component of the U.S. economy, underlying the operations of all businesses, public safety, organizations, and government."
    },
    {
      "value": "manufacturing",
      "expanded": "(CRITICAL MANUFACTURING) The U.S. Department of Homeland Security is designated as the Sector-Specific Agency (SSA) for the Critical Manufacturing Sector."
    },
    {
      "value": "dams",
      "expanded": "(DAMS) The U.S. Department of Homeland Security is designated as the Sector-Specific Agency (SSA) for the Dams Sector. The Dams Sector comprises dam projects, navigation locks, levees, hurricane barriers, mine tailings impoundments, and other similar water retention and/or control facilities."
    },
    {
      "value": "defense",
      "expanded": "(DEFENSE INDUSTRIAL BASE) The he Defense Industrial Base Sector is the worldwide industrial complex that enables research and development, as well as design, production, delivery, and maintenance of military weapons systems, subsystems, and components or parts, to meet U.S. military requirements."
    },
    {
      "value": "emergency svcs",
      "expanded": "(EMERGENCY SERVICES) The U.S. Department of Homeland Security is designated as the Sector-Specific Agency (SSA) for the Emergency Services Sector through the Federal Emergency Management Agency (FEMA). A system of prevention, preparedness, response, and recovery elements, the Emergency Services Sector represents the nation's first line of defense in the prevention and mitigation of risk from terrorist attacks, manmade incidents, and natural disasters."
    },
    {
      "value": "energy",
      "expanded": "(ENERGY) The U.S. energy infrastructure fuels the economy of the 21st century. The U.S. Department of Energy (DOE) is the Sector-Specific Agency (SSA) for the Energy Sector."
    },
    {
      "value": "finance",
      "expanded": "(FINANCIAL SERVICES) The U.S. Department of Treasury is designated as the Sector-Specific Agency (SSA) for the Financial Services Sector."
    },
    {
      "value": "food",
      "expanded": "(FOOD AND AGRICULTURE) The U.S. Department of Agriculture and the U.S. Department of Health and Human Services (HHS) are designated as the co-Sector-Specific Agencies (SSA) for the Food and Agriculture Sector."
    },
    {
      "value": "government",
      "expanded": "(GOVERNMENT FACILITIES) The U.S. Department of Homeland Security and the U.S. General Services Administration (GSA) are designated as the co-Sector-Specific Agencies (SSA) for the Government Facilities Sector."
    },
    {
      "value": "health",
      "expanded": "(HEALTHCARE AND PUBLIC HEALTH) The U.S. Department of Health and Human Services (HHS) is designated as the Sector-Specific Agency (SSA) for the Healthcare and Public Health Sector."
    },
    {
      "value": "IT",
      "expanded": "(INFORMATION TECHNOLOGY) The U.S. Department of Homeland Security is designated as the Sector-Specific Agency (SSA) for the Information Technology Sector."
    },
    {
      "value": "nuclear",
      "expanded": "(NUCLEAR REACTORS, MATERIALS AND WASTE) The U.S. Department of Homeland Security is designated as the Sector-Specific Agency (SSA) for the Nuclear Reactors, Materials, and Waste Sector."
    },
    {
      "value": "transportation",
      "expanded": "(TRANSPORTATION SYSTEMS) The U.S. Department of Homeland Security (through the U.S. Transportation Security Agency [TSA])and the U.S. Department of Transportation (DOT) are designated as the co-Sector-Specific Agences (SSA) for the Transportation Systems Sector."
    },
    {
      "value": "water",
      "expanded": "(WATER AND WASTEWATER SYSTEMS) The U.S. Environmental Protection Agency (EPA) is designated as the Sector-Specific Agency (SSA) for the Water and Wastewater Systems Sector."
    },
  ],
  "values": null
}

Next, is the directory structure:

[root@server taxonomies]# ll
total 68
drwxr-x--- 2 root apache 4096 Feb 19 09:10 admiralty-scale
drwxr-x--- 2 root apache 4096 Feb 19 09:10 adversary
drwxr-x--- 2 root apache 4096 Feb 21 16:59 cipsectors
drwxr-x--- 2 root apache 4096 Feb 19 09:10 circl
drwxr-x--- 2 root apache 4096 Feb 19 09:10 de-vs
drwxr-x--- 2 root apache 4096 Feb 19 09:10 dni-ism
drwxr-x--- 2 root apache 4096 Feb 19 09:10 ecsirt
drwxr-x--- 2 root apache 4096 Feb 19 09:10 euci
drwxr-x--- 2 root apache 4096 Feb 19 09:10 first_csirt_case_classification
drwxr-x--- 2 root apache 4096 Feb 19 09:10 malware
drwxr-x--- 2 root apache 4096 Feb 19 09:10 nato
drwxr-x--- 2 root apache 4096 Feb 19 09:10 osint
-rw-r----- 1 root apache 5456 Feb 19 09:10 README.md
drwxr-x--- 2 root apache 4096 Feb 19 09:10 tlp
drwxr-x--- 4 root apache 4096 Feb 21 13:36 tools
drwxr-x--- 2 root apache 4096 Feb 19 09:10 veris
[root@server taxonomies]# cd cipsectors

**Comment by infracritical** [comment]
I selected "List taxonomies" under "Event Actions", which takes me to the screen containing the list of taxonomies.  On the left side, I click on "Update Taxonomies", and nothing happens.

I would like to point out that when I hover over the clickpoint, it shows "http://1.1.1.1:1111/taxonomies/index#".  Note that I redacted the IP address and service port.  But it does show it as "index#" which doesn't appear to be correct.

I did download the most recent version of MISP,

**Comment by infracritical** [comment]
I figured it out (again)!

But...here's the weird thing that happened...

I found the tools' file, and had originally added the new taxonomy as "cipsector", both with its own directory (and "machinetag.json" file), and to the array within the "./tools/machinetag.py" file.

It originally did *not* add in the new taxonomy when I selected "Update Taxonomies".

However...

When I added it in as "cipsectors" (plural), and corrected the array within the "./tools/machinetag.py" file, somehow -- it worked!

Now...how about that for weird???  Am closing the incident...again!

-rad

P.S.  I hope you're *not* annoyed by my comments or requests...
